vis_gen.py

```diff
@@ -0,0 +1,565 @@
+            num_test: Int, total number of test records
+
+        Raises:
+
+        """
+        # Get absolute path to train and test data directory
+        train_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        test_dir_abs = os.path.join(os.getcwd(), self.out_dir + '_test')
+
+        # Create test dir
+        self.__check_create_out_dir_abs(test_dir_abs)
+
+        # Get total number of training records
+        num_total = len([name for name in os.listdir(train_dir_abs)
+                         if os.path.isfile(os.path.join(train_dir_abs, name))])
+        num_exist = len([name for name in os.listdir(test_dir_abs)
+                         if os.path.isfile(os.path.join(test_dir_abs, name))])
+        if num_total == 0:
+            raise OSError('No data found in specified out_dir.')
+        if num_test > num_total:
+            raise ValueError('Expect num_test to be smaller than total number '
+                             'of records.')
+        if num_exist != 0:
+            raise OSError('Test data already exists.')
+        num_train = num_total - num_test
+
+        # Do train/test split
+        print('Creating train test split with {} total records...'
+              .format(num_total))
+        print('Train size: {}'.format(num_train))
+        print('Test size: {}'.format(num_test))
+        filenames = random.sample(os.listdir(train_dir_abs), 100)
+        for filename in  filenames:
+            srcpath = os.path.join(train_dir_abs, filename)
+            shutil.move(srcpath, test_dir_abs)
+        print('Train test split successfully created.')
+
+        # Check number of classes in each split
+        class_train = set([name.split('_')[0] for name in
+                           os.listdir(train_dir_abs)])
+        class_test = set([name.split('_')[0] for name in
+                          os.listdir(test_dir_abs)])
+        no_missing_class = class_test.issubset(class_train)
+        print('Training dataset has {} categories.'.format(len(class_train)))
+        print('Test dataset has {} categories.'.format(len(class_test)))
+        print('All test categories in training data: {}'
+              .format(no_missing_class))
+
+        return num_train, num_test
+
+
+
+    def __get_filename(self, code_point):
+        """Get the filename for code point under current context. Filename is
+        'CODEPOINT_FONTNAME_FONTSTYLE_ANTIALIAS'
+
+        Args:
+            code_point: Single Unicode code point
+
+        Returns:
+            filename: Str, corresponding file name
+        """
+        filename = 'U+' + str(hex(ord(code_point)))[2:]
+        filename += '_'
+        filename += self.__font_name
+        filename += '_'
+        if self.__font_style:
+            filename += self.font_style
+            filename += '_'
+        filename += self.antialias
+        filename += '.png'
+        return filename
+
+
+    def __check_create_out_dir_abs(self, out_dir_abs):
```

Comment:
single _ prefix for private function per https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations